src/App.tsx

```diff
@@ -1,24 +1,195 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { Todo } from './types/Todo';
+import { Category } from './utils/Category';
+import { Error } from './utils/Error';
 import { UserWarning } from './UserWarning';
+import { TodoForm } from './Components/TodoForm';
+import { TodoList } from './Components/TodoList';
+import { TodoFilter } from './Components/TodoFilter';
+import { addTodo, getTodos, removeTodo } from './api/todos';
 
-const USER_ID = 0;
+const USER_ID = 10156;
 
 export const App: React.FC = () => {
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [isError, setIsError] = useState<Error | string>('');
+  const [filterCategory, setFilterCategory] = useState(Category.All);
+  const [newTodo, setNewTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const todosAmount = visibleTodos.length;
+  const hasCompleted = visibleTodos.some(todo => todo.completed);
+
+  const handleError = (error: Error) => {
+    setIsError(error);
+    window.setTimeout(() => {
+      setIsError('');
+    }, 3000);
+  };
+
+  const loadTodos = async () => {
+    try {
+      const todos = await getTodos(USER_ID);
+
+      setVisibleTodos(todos);
+    } catch {
+      handleError(Error.LOAD);
+    }
+  };
+
+  const filterTodos = async (category: Category) => {
+    const allTodos = await getTodos(USER_ID);
+
+    const filteredTodos = allTodos.filter(todo => {
+      switch (category) {
+        case Category.Active:
+          return !todo.completed;
+        case Category.Completed:
+          return todo.completed;
+        default:
+          return true;
+      }
+    });
+
+    setVisibleTodos(filteredTodos);
+  };
+
+  useEffect(() => {
+    filterTodos(filterCategory);
+  }, [filterCategory]);
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+
+  const handleAdd = async () => {
+    if (!todoTitle) {
+      handleError(Error.TITLE);
+
+      return;
+    }
+
+    try {
+      const todo: Todo = {
+        title: todoTitle,
+        id: 0,
```

Comment:
I think is no need to hardcode here
the id will be generated on the server side